app/Models/User.php

```diff
@@ -860,7 +872,7 @@ public function hasProfile()
     {
         return
             $this->user_id !== null
-            && !$this->isRestricted()
+            && $this->canView()
```

Comment:
I'd separate `hasProfile` to simply mean "is it actual user or bot/guest" and do separate call for authorization.